core/src/main/java/org/infinispan/configuration/cache/TransactionConfigurationBuilder.java

```diff
@@ -327,6 +332,18 @@ public void validate() {
       if (attributes.attribute(TRANSACTION_MODE).get() == TransactionMode.TRANSACTIONAL && !cacheMode.isSynchronous()) {
          throw log.unsupportedAsyncCacheMode(cacheMode);
       }
+      if (attributes.attribute(TRANSACTION_MODE).get() == TransactionMode.TRANSACTIONAL &&
+            attributes.attribute(LOCKING_MODE).get() == LockingMode.OPTIMISTIC &&
+            !attributes.attribute(USE_EAGER_LOCKING).get() && //this is pessimistic transaction
+            getBuilder().locking().isolationLevel() == IsolationLevel.REPEATABLE_READ &&
+            cacheMode != CacheMode.INVALIDATION_SYNC && //write-skew only for REPL, DIST and LOCAL (async is not supported in tx mode)
```

Comment:
invalidation with tx is a weird thing. it was supported, so I keep it. Since it is invalidation mode, we can support write-skew check.